src/main/java/seedu/address/logic/parser/FindCommandParser.java

```diff
@@ -24,10 +33,28 @@ public FindCommand parse(String args) throws ParseException {
             throw new ParseException(
                     String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
         }
+        ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(args, PREFIX_NAME,
+                PREFIX_FINANCIAL_PLAN, PREFIX_TAG);
 
-        String[] nameKeywords = trimmedArgs.split("\\s+");
+        List<String> nameKeywords = argMultimap.getAllValues(PREFIX_NAME);
+        nameKeywords.replaceAll(name -> name.trim());
+        validateNames(nameKeywords);
 
-        return new FindCommand(new NameContainsKeywordsPredicate(Arrays.asList(nameKeywords)));
+        List<String> tagKeywords = argMultimap.getAllValues(PREFIX_TAG);
+        tagKeywords.replaceAll(tag -> tag.trim());
+        validateTags(tagKeywords);
+
+        List<String> financialPlanKeywords = argMultimap.getAllValues(PREFIX_FINANCIAL_PLAN);
+        financialPlanKeywords.replaceAll(financialPlan -> financialPlan.trim());
+        validateFinancialPlans(financialPlanKeywords);
+
+        CombinedPredicate combinedPredicate = new CombinedPredicate(
+                new FinancialPlanContainsKeywordsPredicate(financialPlanKeywords),
+                new NameContainsKeywordsPredicate(nameKeywords),
+                new TagContainsKeywordsPredicate(tagKeywords)
+        );
+
+        return new FindCommand(combinedPredicate);
```

Comment:
Parser allows for incorrect format of find to be entered ie. if i enter `find john` it just executes the command and returns nothing instead of returning an issue with using the function since you need to do `find n/john`. Not sure if this is a design choice but it differs slightly from the way gather is implemented in that gather will not allow you to execute the command if its not `gather /fp financialplan`

Reply:
Yeah that's not intended. I'll fix it.